Source/WebCore/accessibility/AccessibilityObject.cpp

```diff
@@ -3453,7 +3458,12 @@ bool AccessibilityObject::isExpanded() const
     }
 
     if (supportsExpanded()) {
-        if (RefPtr popoverTargetElement = this->popoverTargetElement())
+        if (RefPtr invokeTargetElement = this->invokeTargetElement()) {
+            if (invokeTargetElement->isPopoverShowing())
+                return true;
+            if (RefPtr dialog = dynamicDowncast<HTMLDialogElement>(invokeTargetElement))
+                return dialog->isOpen();
```

Comment:
```suggestion
```

We likely don't want expanded on Dialog invokers as they're only going to show as modal, so the information about the expanded state is a little redundant (you know if a modal dialog is expanded because you'll be in it).